letsApply/Controllers/TabBarViewController/HomeScreenViewController/HomeViewController/HomeScreenViewController.swift

```diff
@@ -5,217 +5,174 @@
         
-        NSLayoutConstraint.activate([
-            searchBar.topAnchor.constraint(equalTo: searchBarContainerView.topAnchor),
-            searchBar.leadingAnchor.constraint(equalTo: searchBarContainerView.leadingAnchor),
-            searchBar.trailingAnchor.constraint(equalTo: searchBarContainerView.trailingAnchor),
-            searchBar.bottomAnchor.constraint(equalTo: searchBarContainerView.bottomAnchor)
-        ])
+        jobsCollectionView.topAnchor.constraint(equalTo: categoriesCollectionView.bottomAnchor, constant: 20).isActive = true
+        jobsCollectionView.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 20).isActive = true
+        jobsCollectionView.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -20).isActive = true
+        jobsCollectionView.bottomAnchor.constraint(equalTo: view.bottomAnchor).isActive = true
         
-        // Metrics CollectionView
-        NSLayoutConstraint.activate([
-            metricsCollectionView.topAnchor.constraint(equalTo: searchBarContainerView.bottomAnchor, constant: 10),
-            metricsCollectionView.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 10),
-            metricsCollectionView.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -10),
-            metricsCollectionView.heightAnchor.constraint(equalToConstant: 120)
-        ])
     }
     
-    private func setupViewModel() {
-        viewModel.onDataUpdated = { [weak self] in
-            DispatchQueue.main.async {
-                self?.updateUI()
-            }
-        }
+    private func setupNavigationBar() {
+        navigationItem.searchController = searchController
+        navigationItem.hidesSearchBarWhenScrolling = false
     }
     
-    private func fetchUserProfile() {
-        if let uid = Auth.auth().currentUser?.uid {
-            viewModel.fetchUserProfile(uid: uid)
-        }
-    }
+    // MARK: - Load Data
     
-    private func updateUI() {
-        if let profile = viewModel.userProfile {
-            greetingLabel.text = "Hey, \(profile.name) 👋"
-            
-            if let profileUrl = profile.profilePictureUrl {
-                ProfilePictureService.shared.fetchProfilePicture(urlString: profileUrl) { [weak self] image in
-                    self?.profileImageView.image = image
-                }
-            } else {
-                profileImageView.image = UIImage(systemName: "person.circle.fill")
+    private func loadUserProfile() {
+        homeViewModel.fetchUserProfile { [weak self] profile in
+            self?.greetingLabel.text = "Hello, \(profile.name)!"
+            self?.homeViewModel.loadProfileImage(urlString: profile.profilePictureUrl) { image in
+                self?.profileImageView.image = image
+            }
+            self?.homeViewModel.fetchRelevantJobs(for: profile) { jobs in
+                self?.jobs = jobs
+                self?.jobsCollectionView.reloadData()
```

Comment:
Don't forget to move these operations to the main thread. Otherwise you will be updating your UI on a background thread.